src/validation.cairo

```diff
@@ -70,9 +70,24 @@ fn validate_merkle_root(self: @ChainState, block: @Block) -> Result<(), ByteArra
     Result::Ok(())
 }
 
+fn compute_block_reward(self: @ChainState, block: @Block) -> Result<felt252, ByteArray> {
+    let number_halvings: u32 = *self.block_height / 210000;
+    match number_halvings {
+        0 => { return Result::Err("number_halvings equal 0"); },
+        _ => {}
+    }
+    let denominator: u32 = number_halvings*number_halvings;
```

Comment:
It's actually `2**number)halvings`

You can use this snippet if you'd like https://github.com/m-kus/raito/pull/1/files#diff-6738ab9bc884db610afe82537c430edd6398401462313df53e93d7121b812964R128

Reply:
Appreciate thank you! I've made few changes